src/source/Crypto/Tls_mbedtls.c

```diff
@@ -114,9 +114,25 @@ STATUS tlsSessionStart(PTlsSession pTlsSession, BOOL isServer)
         STATUS_CREATE_SSL_FAILED);
 
     mbedtls_ssl_conf_ca_chain(&pTlsSession->sslCtxConfig, &pTlsSession->cacert, NULL);
-    mbedtls_ssl_conf_authmode(&pTlsSession->sslCtxConfig, MBEDTLS_SSL_VERIFY_REQUIRED);
+
+    // For mbedTLS 3.x, use appropriate verification mode
+    if (hostname != NULL) {
+        // If a hostname is provided, use strict verification
+        mbedtls_ssl_conf_authmode(&pTlsSession->sslCtxConfig, MBEDTLS_SSL_VERIFY_REQUIRED);
+    } else {
+        // Otherwise, use optional verification for IP-based connections
+        mbedtls_ssl_conf_authmode(&pTlsSession->sslCtxConfig, MBEDTLS_SSL_VERIFY_OPTIONAL);
```

Comment:
In the current code, it seems to always choose optional verify (added a print to confirm), so verify required path never gets run.

After fixing the hostname assignment:
```c
PCHAR hostname = MEMCALLOC(1, STRLEN(pTurnServer->url) + 1);
CHK(hostname != NULL, STATUS_NOT_ENOUGH_MEMORY);
SNPRINTF(hostname, STRLEN(pTurnServer->url) + 1, "%s", pTurnServer->url);
pTurnSocket->hostname = hostname;
```

It will choose the verify required path.

Reply:
That's right, so if we have hostname provided, set this as `MBEDTLS_SSL_VERIFY_REQUIRED`.
Please note that for mbedtls 3.6.3 and later and the latest 2.8.x release, the `REQUIRED` option is by default.

Reply:
Ohh! You mean the hostname is not getting passed as expected! Wondering why that would happen! 🤔

Reply:
In the previous logic, if the string didn't have `://` in it, the output wouldn't get set (but still returns success). Looks like the issue is resolved in the latest.